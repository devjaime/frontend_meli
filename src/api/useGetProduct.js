import axios from 'axios';
import { useEffect, useState } from 'react';

const API_BASE = process.env.REACT_APP_API_BASE;

export const useGetProduct = (itemId) => {
  const [response, setResponse] = useState(null);
  const [error, setError] = useState(null);
  useEffect(() => {
    let isSubscribed = true;
    const fetchData = async () => {
      try {
        const options = {
          'Content-Type': 'application/json',
        };
        const res = await axios.get(
          `${API_BASE}/api/items/${itemId}`,
          options,
        );
        setResponse(await res.data);
        if (isSubscribed) setResponse(response);
      } catch (err) {
        setError(err);
      }
    };
    if (isSubscribed) fetchData().then();
    isSubscribed = false;
    return () => (isSubscribed);
  }, [itemId]);
  return { response, error };
};
export default useGetProduct;
