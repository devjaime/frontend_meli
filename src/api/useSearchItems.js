import axios from 'axios';
import { useEffect, useState } from 'react';

const api_base = process.env.REACT_APP_API_BASE;

const useSearchItems = query => {
    const [response, setResponse] = useState(null);
    const [error, setError] = useState(null);
    useEffect(() => {
        let isSubscribed = true;
        const fetchData = async () => {
            try {
                let options = { 
                    'Content-Type': 'application/json',
                    params: {
                        q: query
                    }
                }
           
                const res = await axios.get(
                    `${api_base}/api/items`,
                    options
                );
                console.log(JSON.stringify(res.data))
                const data = await res.data;
                
                if (isSubscribed) setResponse(data);
            } catch (error) {
                console.error('something happened', error);
                setError(error);
            }
        }
        if (isSubscribed) fetchData().then();
        return () => (isSubscribed = false);
    }, [query]);
    return { response, error };
}

export default useSearchItems;