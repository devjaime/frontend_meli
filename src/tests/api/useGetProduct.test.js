import React from 'react';
import { shallow } from "enzyme";
import { useGetProduct } from '../../api/useGetProduct';

describe('Test in useGetProduct', () => {
    const itemId='MLA902105833'
    test('should display the component correctly', () => {
        const wrapper =  shallow( <useGetProduct itemId={itemId} />)
        expect(wrapper).toMatchSnapshot();
    })
})