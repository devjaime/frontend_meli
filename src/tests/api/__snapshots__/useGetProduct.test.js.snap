// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in useGetProduct should display the component correctly 1`] = `
<useGetProduct
  itemId="MLA902105833"
/>
`;
